### ISSUE TITLE
Batch predictions Image Captioning task

### ISSUE THREAD

**Issue by MikeMACintosh** [salesforce/BLIP#58]
Hi, glad to see and you this cool project, thanks you.
I have a questions: if it possible to batch predictions on Image captioning task?
I see https://github.com/salesforce/BLIP/issues/48 but it's not my case.

i do something like:

base_model_path = 'path_to_base_model'
model_base: str = blip_decoder(pretrained=base_model_path, vit='base', image_size=IMAGE_SIZE) 
model_base.eval()
model_base.to(device)

img = transform(sample).unsqueeze(0).to(device)
with torch.no_grad():
    caption_bs_base=model_base.generate(img, sample=False, num_beams=7, max_length=16, min_length=5)

It works good, but i want to inference 4 models(vit base/large and beam search/nucleus sampling) and it's to long. On my server signature 12 pictures 4 models takes ~43 sec.

Thanks you.

**Comment by LiJunnan1992** [comment]
Yes you can do batch inference.

**Comment by MikeMACintosh** [comment]
@LiJunnan1992  Сould you explain how i can do that? Should I write my own Dataloader?

**Comment by poipiii** [comment]
yes you have to write your own data loader I just done it myself

**Comment by shams2023** [comment]
> Hi, glad to see and use this cool project, thanks you. I have a question: if it possible to batch predictions on Image captioning task? I see #48 but it's not my case.
> 
> i do something like:
> 
> base_model_path = 'path_to_base_model' model_base = blip_decoder(pretrained=base_model_path, vit='base', image_size=IMAGE_SIZE) model_base.eval() model_base.to(device)
> 
> img = transform(sample).unsqueeze(0).to(device) with torch.no_grad(): caption_bs_base=model_base.generate(img, sample=False, num_beams=7, max_length=16, min_length=5)
> 
> It works good, but i want to inference 4 models(vit base/large and beam search/nucleus sampling) and it's to long. On my server signature 12 pictures 4 models takes ~34 sec (12*4 = 48 signature).
> 
> Thanks you.



> Hi, glad to see and use this cool project, thanks you. I have a question: if it possible to batch predictions on Image captioning task? I see #48 but it's not my case.
> 
> i do something like:
> 
> base_model_path = 'path_to_base_model' model_base = blip_decoder(pretrained=base_model_path, vit='base', image_size=IMAGE_SIZE) model_base.eval() model_base.to(device)
> 
> img = transform(sample).unsqueeze(0).to(device) with torch.no_grad(): caption_bs_base=model_base.generate(img, sample=False, num_beams=7, max_length=16, min_length=5)
> 
> It works good, but i want to inference 4 models(vit base/large and beam search/nucleus sampling) and it's to long. On my server signature 12 pictures 4 models takes ~34 sec (12*4 = 48 signature).
> 
> Thanks you.

Excuse me, have you written it out yet?
I also want to process my own image data set (i.e. generate caption), how to operate, I hope to get some knowledge from you to help me solve this problem.

**Comment by shams2023** [comment]
> 是的，您必须编写自己的数据加载器，我只是自己完成的

Hello, big man!
Can I ask you for a copy of this dataloder you wrote? I also want to use BLIP to caption their own data set, but I have been writing out of the problem, I want you to consult a can run, thank you!
